e2e/tests/e2e_happy_path/HappyPath.spec.ts

```diff
@@ -31,65 +34,119 @@ const editor: Editor = e2eContainer.get(CLASSES.Editor);
-        await ide.closeAllNotifications();
-        await topMenu.clickOnTopMenuButton('Terminal');
-        await topMenu.clickOnSubmenuItem('Run Task...');
+    });
+
+    test('Build application', async () => {
+        await topMenu.selectOption('Terminal', 'Run Task...');
         await quickOpenContainer.clickOnContainerItem('che: build-file-output');
 
         await projectTree.expandPathAndOpenFile(projectName, 'build-output.txt');
-        await editor.waitEditorAvailable('build-output.txt');
-        await editor.clickOnTab('build-output.txt');
-        await editor.waitTabFocused('build-output.txt');
         await editor.followAndWaitForText('build-output.txt', '[INFO] BUILD SUCCESS', 180000, 5000);
     });
 
     test('Run application', async () => {
-        await topMenu.waitTopMenu();
-        await ide.closeAllNotifications();
-        await topMenu.clickOnTopMenuButton('Terminal');
-        await topMenu.clickOnSubmenuItem('Run Task...');
+        await topMenu.selectOption('Terminal', 'Run Task...');
         await quickOpenContainer.clickOnContainerItem('che: run');
 
-        await ide.waitNotification('A new process is now listening on port 8080', 120000);
-        await ide.clickOnNotificationButton('A new process is now listening on port 8080', 'yes');
+        await ide.waitNotificationAndConfirm('A new process is now listening on port 8080', 120000);
+        await ide.waitNotificationAndOpenLink('Redirect is now enabled on port 8080', 120000);
+    });
+
+    test('Check the running application', async () => {
+        await previewWidget.waitContentAvailable(SpringAppLocators.springTitleLocator, 60000, 10000);
+    });
 
-        await ide.waitNotification('Redirect is now enabled on port 8080', 120000);
-        await ide.clickOnNotificationButton('Redirect is now enabled on port 8080', 'Open Link');
-        await previewWidget.waitContentAvailable(springTitleLocator, 60000, 10000);
+    test('Close preview widget', async () => {
         await rightToolbar.clickOnToolIcon('Preview');
         await previewWidget.waitPreviewWidgetAbsence();
     });
 
+    test('Close the terminal running tasks', async () => {
+        await terminal.closeTerminalTab('build-file-output');
+        await terminal.rejectTerminalProcess('run');
+        await terminal.closeTerminalTab('run');
+
+        await warningDialog.waitAndCloseIfAppear();
+    });
+});
+
+suite('Language server validation', async () => {
     test('Java LS initialization', async () => {
         await projectTree.expandPathAndOpenFile(pathToJavaFolder, javaFileName);
-        await editor.waitEditorAvailable(javaFileName);
-        await editor.clickOnTab(javaFileName);
-        await editor.waitTabFocused(javaFileName);
+        await editor.selectTab(javaFileName);
+
+        await ide.checkLsInitializationStart('Starting Java Language Server');
         await ide.waitStatusBarTextAbcence('Starting Java Language Server', 360000);
```

Comment:
I know this is out of scope of this PR, but for some next time - `Abcence` doesn't exist - it's `Absence`